README.md

```diff
@@ -1,5 +1,88 @@
 # How to run
-
-Run ```docker run -p 5432:5432 -e POSTGRES_PASSWORD=password -e POSTGRES_USER=user postgres``` before starting 
+Run ```docker run -p 5432:5432 -e POSTGRES_PASSWORD=password -e POSTGRES_USER=user postgres``` before starting
 the main Spring Boot application.
+# What is the application?
+REST API monolith with a postgres database. The application.yml
+    contains the app's configuration details, like Liquibase config for database
+    migration.
+
+### Endpoints
+
+1. **Get all customers**
+    - **Request:**
+        - Method: GET
+        - URL: `/api/v1/customers`
+        - Example Request:
+          ```
+          GET /api/v1/customers
+          ```
+
+    - **Response:**
+        - Status Code: 200 OK
+        - Body: Array of customer objects containing `id`, `name`, `email`, and `age` fields.
+
+2. **Add a new customer**
+    - **Request:**
+        - Method: POST
+        - URL: `/api/v1/customers`
+        - Example Request:
+          ```
+          POST /api/v1/customers
+          Content-Type: application/json
+   
+          {
+            "name": "John Doe",
+            "email": "john@example.com",
+            "age": 30
+          }
+          ```
+
+    - **Response:**
+        - Status Code: 201 Created
+        - Body: New Customer object containing `id`, `name`, `email`, and `age` fields.
+
+3. **Delete a customer**
+    - **Request:**
+        - Method: DELETE
+        - URL: `/api/v1/customers/{customerId}`
+        - Example Request:
+          ```
+          DELETE /api/v1/customers/123
+          ```
+
+    - **Response:**
+        - Status Code: 200 OK
+        - Body: Success message confirming deletion.
+
+4. **Update a customer**
+    - **Request:**
+        - Method: PUT
+        - URL: `/api/v1/customers/{customerId}`
+        - Example Request:
+          ```
+          PUT /api/v1/customers/123
+          Content-Type: application/json
+   
+          {
+            "name": "Updated Name",
+            "email": "updated@example.com",
+            "age": 35
+          }
+          ```
+
+    - **Response:**
+        - Status Code: 201 Created
+        - Body: New Customer object containing `id`, `name`, `email`, and `age` fields.
+
+
+
+### How the Test works
+   **gg**
+ - The integration test uses PostgreSQLContainer to spin up a Docker containerised PostgreSQL database instance.This allows the tests to run against a real database without affecting any existing databases.
+ - The `Initializer` class implements the ApplicationContextInitializer interface which allows us to override the app properties related to the db connection with properties of PostgreSQL container instance.
+   This allows the App to connect to the PostgreSQL container when the test is run.
+   
+- JDBI is used to interact with the PostgreSQL database in the test, including adding/deleting customers
+
```

Comment:
There are a few things missing here and some sentences don't make sense, "tests to run against a real database without affecting any existing databases". You haven't mentioned Liquibase at all.

You need to mention something like "Since this is a standard Java web application the integration test spins up the controller, service and database layer. As part of the test it uses the [TestContainers](https://testcontainers.com/) library to setup a Postgres container."

Then mention the `CustomerControllerTest.Initializer` static class overrides the application properties in `application.yml` so that our application points to the Postgres container.

Next liquibase runs the database migrations so that the Postgres container contains a `Customer` table.

Next JDBI connects to our Postgres container to add test data into our Postgres container (e.g. adding and removing customers) for our tests to work. An `@AfterEach` hook runs after each test to clear away the test data.